pegasus/src/homepage.rb

```diff
@@ -1,6 +1,71 @@
 class Homepage
   MAX_HEROES_IN_ROTATION = 6
 
+  @@announcements_data = nil
+  @@loaded = false
+  @@load_error = false
+  @@json_path = pegasus_dir 'sites.v3/code.org/homepage.json'
+
+  # gets special announcement data for a page, or nil if not found
+  def self.get_announcement_for_page(page)
+    load_announcements
+    return nil if @@load_error || !@@announcements_data
+    pages = @@announcements_data[:pages]
+    banners = @@announcements_data[:banners]
+    banner_id_for_page = pages[page]
+    return nil unless banner_id_for_page
+
+    banner = banners[banner_id_for_page]
+
+    # If the banner has an array of environments, then the current environment must be one of them.
+    return nil if banner["environments"] && !banner["environments"].include?(CDO.rack_env.to_s)
+
+    # If the banner has a required DCDO flag, then it must be set.
+    return nil if banner["dcdo"] && !DCDO.get(banner["dcdo"], false)
+
+    banner ? banner.merge({"id": banner_id_for_page}) : nil
+  end
+
+  def self.load_announcements
+    # Reloads JSON file with announcement data on each page load
+    # in non-production environments
+    unless (@@load_error || @@loaded) && rack_env?(:production)
```

Comment:
@molly-moen I think I need to remove the `&& rack_env?(:production)` if I want to be able to change the announcement via DCDO on production?

Reply:
Yeah I think so. If it's not there then I believe it will only be changeable on a dtp.